logging.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type requestLog struct {
+	URL     string
+	Method  string
+	Headers map[string][]string
+	Body    string
+}
+
+type responseLog struct {
+	Status  int
+	Headers map[string][]string
+	Body    string
+}
+
+func logProxyRequest(r *http.Request) {
+	b, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Failed to read body from request: %s", err)
+		return
+	}
+
+	r.Body.Close()
```

Comment:
I think the body will never get closed if `err != nil` resulting in a leaked file descriptor.

I'd recommend putting this at the top of your function:

```Go
func logProxyRequest(r *http.Request) {
  defer r.Body.Close()
```